File: CS/DXExample.Module/DomainObject1.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace DXExample.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private string _Description;
        public string Description {
            get { return _Description; }
            set { SetPropertyValue("Description", ref _Description, value); }
        }
        private string _HiddenProperty;
        [RemoveFromViewInfo]
        public string HiddenProperty {
            get { return _HiddenProperty; }
            set { SetPropertyValue("HiddenProperty", ref _HiddenProperty, value); }
        }
    }
}

File: VB/DXExample.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace DXExample.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
		Private _Description As String
		Public Property Description() As String
			Get
				Return _Description
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Description", _Description, value)
			End Set
		End Property
		Private _HiddenProperty As String
		<RemoveFromViewModel> _
		Public Property HiddenProperty() As String
			Get
				Return _HiddenProperty
			End Get
			Set(ByVal value As String)
				SetPropertyValue("HiddenProperty", _HiddenProperty, value)
			End Set
		End Property
	End Class
End Namespace
